icechunk-python/python/icechunk/__init__.py

```diff
@@ -161,6 +166,18 @@ def __setstate__(self, state: Any) -> None:
         state["_store"] = pyicechunk_store_from_bytes(store_repr, read_only)
         self.__dict__ = state
 
+    @contextlib.contextmanager
+    def preserve_read_only(self) -> Generator[None, None, None]:
+        """
+        Context manager to allow unpickling this store preserving `read_only` status.
+        By default, stores are set to read-only after unpickling.
+        """
+        try:
+            self._pickle_preserves_read_only = True
+            yield
+        finally:
+            self._pickle_preserves_read_only = False
```

Comment:
shouldn't this be the previous value of the variable instead?

Reply:
it's always only modified in this method, so the previous value is always False.